homeassistant/components/advantage_air/climate.py

```diff
@@ -117,6 +120,16 @@ def __init__(self, instance: AdvantageAirData, ac_key: str) -> None:
             # MyZone
             self._attr_supported_features |= ClimateEntityFeature.TARGET_TEMPERATURE
 
+        # Add "MyTemp" preset if available
+        if ADVANTAGE_AIR_MYTEMP_ENABLED in self._ac:
+            self._attr_preset_modes += [ADVANTAGE_AIR_MYTEMP]
+            self._attr_supported_features |= ClimateEntityFeature.PRESET_MODE
+
+        # Add "MyAuto" preset if available
+        if ADVANTAGE_AIR_MYAUTO_ENABLED in self._ac:
+            self._attr_preset_modes += [ADVANTAGE_AIR_MYAUTO]
```

Comment:
Why is myauto both an HVAC mode and a preset?

Reply:
The products "auto" mode happens to be called "myauto" and is only available when in the MyAuto preset.

Reply:
Are you sure that it's really two different things? In the pdf you linked, talking about myauto they say:

> Once this feature is enabled you can select it on the main mode page.

Reply:
They have two completely seperate UI selections, and are two different parts of the API.

Reply:
How can the myauto preset functionality work with another mode selected. It should automatically select heating or cooling.

Reply:
It automatically heats and cools only when set to "Auto" which is called heat_cool in home assistant, and this HVAC mode is only available in the "MyAuto" preset.

The MyAuto preset has other differences than just the HVAC mode, like it uses the average temperature from all zones rather than just one zone in MyZone.

All of this functionality is already in the integration in Home Assistant, I just need this PR approved so users can _change_ the temperature control mode (preset), like those on the HACS version have been able to for almost a year now.

Reply:
Have you confirmed that other combinations than mode myauto and preset myauto actually work and have those combined effects?

Reply:
No I have not confirmed that personally because my house only supports MyTemp. However I have users in the community which have been using this implementation through HACS for almost a year, and have shared their diag's so I can implement this feature for them.

I also dont see how this is relevant to the PR.

Reply:
It's relevant as we're implementing a feature that currently enables those combinations. If they don't work we shouldn't do it like that.

Reply:
Those features were implemented in https://github.com/home-assistant/core/pull/91107 and then fixed slightly in https://github.com/home-assistant/core/pull/93189